Program.cs

```diff
@@ -145,6 +146,7 @@ private void registerMiddleware(ContainerBuilder containerBuilder)
             _logger.LogInformation("Reading middleware configuration...");
             var middlewareConfigRoot = Configuration.GetSection("Middleware");
             var middlewareConfigs = middlewareConfigRoot.GetChildren();
+            var middlewareMetadata = new List<MiddlewareMetadata>();
```

Comment:
I'm not sure we necessarily need to store this information, it should all be available at runtime via `IEnumerable<IMiddleware>`

Reply:
If I give the middleware access to just the IEnumerable<IMiddleware> then it would have to at least once (but can potentially continuously) also interact with the application config and run reflection stuff with specific interfaces against other middlewares to access the same information.

I designed this with the goal in mind to separate these concerns and have the pipeline provide these values to the middleware so it will never have worry about the appconfig or internal implementation details.